packages/auth/src/accessToken/service.ts

```diff
@@ -65,3 +75,27 @@ async function introspect(
     return { active: true, ...grant, key: { proof: 'httpsig', jwk: keys[0] } }
   }
 }
+
+async function revoke(
+  deps: ServiceDependencies,
+  id: string
+): Promise<RevocationResult> {
+  const token = await AccessToken.query(deps.knex).findOne({ id })
+  if (token) {
+    if (!isTokenExpired(token)) {
+      AccessToken.query(deps.knex)
+        .update({
+          expiresIn: 1
+        })
+        .where('id', token.id)
+    }
+
+    return {
+      foundToken: true
+    }
+  } else {
+    return {
+      foundToken: false
+    }
```

Comment:
I wonder if that should return an Error instead of the object or just `false`.

Reply:
changed to return `Error | undefined`